doc/design/logs.md

```diff
@@ -0,0 +1,35 @@
+# Logs in SQLFlow
+## Motivation
+In order to know well about the runtime status of the SQLFlow job, we need to count the number of `TRAIN/PREDICT/EXPLAIN/NORMAL` tasks over a period of time. For example, using ELK Stack for log query and analysis‎. Generally, such statistics components are implemented by logs.
+
+## logging library
+Several statistics components use `logtail` to collect logs. The `logtail` demands a file. If such information is written to `stdout`, we should redirect the `stdout` to a local file, the file size will increase until the SQLFlow server terminates, so we need a rolling file to store the log. Besides, a well-formatted log is readable.
+We use [Logrus](https://github.com/sirupsen/logrus) for layout, combining with [https://github.com/natefinch/lumberjack] for rolling the log file.
+
+## Log Formatter
+Logs in their raw form are typically a text format with one event per line, i.e. :    
+`2020-03-10 10:00:14 level={Level} requestID={RequestID} user={UserID} event={event} msg={Metric, Result or Details}`
+
+RequestID: We use workflow ID as request ID if existed, or UUID instead. With the same RequestID, organizing these logs(events) in order can be reduced to a story;
+user: The one who submits the SQL;
+event: The process of the task, such as: parsing;
```

Comment:
Are we only adding the logging in the server package?

If so, there are only two types of requests run/fetch, I am not sure how event maps to them.

If not, how about simply logging the file name, so we can avoid coding the event name like 
```
log.WithFields(log.Fields{
    "event": "parsing",
  }).Info("Done parsing")
```

Reply:
Not only the `server` package need logging, but also other packages.
"..logging the file name..", do you mean we naming the log file according to the package name? I totally agree.

Reply:
> do you mean we naming the log file according to the package name?

Yes.